src/modules/roc_fec/reader.cpp

```diff
@@ -114,6 +114,15 @@ bool Reader::update_size_(size_t new_sblen) {
         return true;
     }
 
+    if (new_sblen > decoder_.max_source_block()) {
```

Comment:
The documentation for max_source_block() states that it can return zero, but there is no check here. We should either fix the documentation or the usage.

If we will use of_get_control_parameter(), we probably can just panic on its error (it should not happen) and fix documentation.

Reply:
Reworked.